### PR TITLE
Fix #14197

### LINKED ISSUE #14197: Allow to create internal share
Currently one can only create public links. Allow to create share [^1] of type `0=user` or `4=email` with internal users. Requires an input field option to select the user registered. Should be possible to lookup other users using the _Sharee API_ [^2].

[^1]: https://docs.nextcloud.com/server/latest/developer_manual/client_apis/OCS/ocs-share-api.html#create-a-new-share
[^2]: https://docs.nextcloud.com/server/latest/developer_manual/client_apis/OCS/ocs-sharee-api.html#search-sharees

### PR DESCRIPTION
**Pull request by dkocher**

### CODE REVIEW

**Comment by AliveDevil** [comment]
For Windows I'd like to clarify following:
- Title of Window is "Share…" (Table "Main")
- Buttons: "Choose" (Table Localizable), "Cancel…" (Table Localizable)
- Is there any prompt/instruction text I can use, to make this less empty?
- Selection box below instruction text
- App-icon left of instruction text

**Comment by AliveDevil** [comment]
One thing I'd like to consider for the future: Make the Share-dialog a wizard.
Now:
- A dialog opens for Sharee-selection
- A password prompt opens
- Share link is copyable in another dialog

This three-step process could just be put into a wizard.

**Comment by AliveDevil** [comment]
Implemented as follows:

image

**Review by ylangisc — COMMENTED** [review]
Consistent naming for all share features → `...ShareFeature`?

File: core/src/main/java/ch/cyberduck/core/shared/DefaulShareFeature.java [hunk 1]
```diff
@@ -15,14 +15,18 @@
...
+import ch.cyberduck.core.DescriptiveUrl;
+import ch.cyberduck.core.PasswordCallback;
+import ch.cyberduck.core.Path;
+import ch.cyberduck.core.UrlProvider;
 import ch.cyberduck.core.exception.BackgroundException;
-import ch.cyberduck.core.features.PromptUrlProvider;
+import ch.cyberduck.core.features.Share;
 
-public class DefaulPromptUrlProvider implements PromptUrlProvider {
+public class DefaulShareFeature implements Share {
```
Comment (ylangisc) [inline core/src/main/java/ch/cyberduck/core/shared/DefaulShareFeature.java:25]: Typo.

**Comment by ylangisc** [comment]
> Consistent naming for all share features → `...ShareFeature`?

Not resolved or commented yet.

**Comment by dkocher** [comment]
> > Consistent naming for all share features → `...ShareFeature`?
> 
> Not resolved or commented yet.

In 0e43674b7eb58c1470bdc85012bcbf150bc744e6. Would keep the class names for other implementations.

**Review by ylangisc — APPROVED** [review]

**Pull request merged**